feat: add test osaka replay test

letypequividelespoubelles opened the pull request:
<!-- CURSOR_SUMMARY -->
> [!NOTE]
> Adds Osaka Sepolia replay tests and resources, switches conflation blobBaseFees to byte[] with default decoding, and introduces a DEVNET chain config while updating runners to use blobBaseFeesOrDefault.
> 
> - **Tests (Osaka/Sepolia)**:
>   - Add `SepoliaReplayTests` with multiple block replays using `SEPOLIA_TESTCONFIG(OSAKA)`.
>   - Add replay resource `replays/osaka/oogModexp.json`.
>   - Remove obsolete `replays/issue-1090/blocks.csv` and `tests.txt`.
> - **Conflation Snapshot**:
>   - Change `ConflationSnapshot.blobBaseFees` type to `Map<Long, byte[]>` and encode in `from(...)`.
>   - Add decoding and defaulting via `blobBaseFeesOrDefault()` using `getDefaultBlobBaseFees(...)`.
> - **Replay/Execution**:
>   - Update `ReplayTestTools` and `MultiBlockExecutionEnvironment` to pass `blobBaseFeesOrDefault()` to `PublicInputs`.
> - **Chain Config**:
>   - Introduce `DEVNET_CHAIN_ID` and `DEVNET_TESTCONFIG(...)`.
>   - Minor API tidy (broader static import, method signatures).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e2ed51765fbc93a552bf2c448b8b5211915c7582. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

letypequividelespoubelles committed:
feat: add test osaka replay test

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles committed:
feat: add test osaka replay test

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles committed:
ras

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles committed:
ras

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles commented on arithmetization/src/test/resources/replays/osaka/oogModexp.json:
```diff
@@ -0,0 +1 @@
+{"jsonrpc":"2.0","id":1,"result":{"capture":"{\"fork\":\"LONDON\",\"blocks\":[{\"header\":{\"parentHash\":\"0x132a1775404b2c42fc89e27043e0a5ad7857fb1dc3b1b7bb436d9a7bf5c811bf\",\"ommersHash\":\"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347\",\"coinbase\":\"0x0000000000000000000000000000000000000000\",\"stateRoot\":\"0x811836250d111695914e7cf47701421c4e6a7f2a1e3f667cac166d606e ...
```
fork == London ?

letypequividelespoubelles committed:
Merge branch 'arith-dev' into 2538-add-one-osaka-replay-tests

letypequividelespoubelles committed:
rm useless files

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles committed:
rm useless files

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles committed:
Merge branch '2538-add-one-osaka-replay-tests' of github.com:Consensys/linea-tracer into 2538-add-one-osaka-replay-tests

DavePearce committed:
apply jq to json file

DavePearce committed:
apply jq to json file

lorenzogentile404 committed:
added partial fix to read JSON sorrounded by double quotes

lorenzogentile404 committed:
added partial fix to read JSON sorrounded by double quotes

lorenzogentile404 committed:
added general fix for handling differen JSON formats

lorenzogentile404 committed:
added general fix for handling differen JSON formats

letypequividelespoubelles committed:
make it cleaner

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles committed:
make it cleaner

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles committed:
Merge branch 'arith-dev' into 2538-add-one-osaka-replay-tests

DavePearce commented on arithmetization/src/test/resources/replays/osaka/oogModexp.json:
```diff
@@ -0,0 +1 @@
+"{\"fork\":\"LONDON\",\"blocks\":[{\"header\":{\"parentHash\":\"0x132a1775404b2c42fc89e27043e0a5ad7857fb1dc3b1b7bb436d9a7bf5c811bf\",\"ommersHash\":\"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347\",\"coinbase\":\"0x0000000000000000000000000000000000000000\",\"stateRoot\":\"0x811836250d111695914e7cf47701421c4e6a7f2a1e3f667cac166d606eb368d8\",\"transactionRoot\":\"0x71ae528cfd0 ...
```
Yes, this looks odd to me.

DavePearce committed:
add capture osaka replays from sepolia

This adds a number of captured replays from sepolia.

DavePearce commented on arithmetization/src/test/resources/replays/osaka/oogModexp.json:
```diff
@@ -0,0 +1 @@
+{"jsonrpc":"2.0","id":1,"result":{"capture":"{\"fork\":\"LONDON\",\"blocks\":[{\"header\":{\"parentHash\":\"0x132a1775404b2c42fc89e27043e0a5ad7857fb1dc3b1b7bb436d9a7bf5c811bf\",\"ommersHash\":\"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347\",\"coinbase\":\"0x0000000000000000000000000000000000000000\",\"stateRoot\":\"0x811836250d111695914e7cf47701421c4e6a7f2a1e3f667cac166d606e ...
```
fixed

DavePearce commented on arithmetization/src/test/java/net/consensys/linea/ReplayTestTools.java:
```diff
@@ -89,8 +91,20 @@ public static void replay(
+    final String conflationAsString;
+    final ConflationSnapshot conflation;
+
+    if (jsonIsString) {
```
This doesn't really make sense to me.   I think its because of that odd replay you had `oogModexp.json` which was malformed.  We should the traces are well-formed.  Note also that they are potentially compressed as well.

DavePearce committed:
remove workaround for malformed replay file

This removes a (broken) workaround for a malformed replay file.  Simpler
solution: ensure replay files are not malformed.

DavePearce committed:
change json encoding of blobBaseFees

This updates the json encoding of blobBaseFees to use a straightforward
array encoding (i.e. byte[]) which can be automatically managed by the
GSON decoder.  This prevents the need to develop a custom decoder for
Bytes, and prevents the replay files from being dependent upon the
internal representation of this class.

DavePearce committed:
recapture replays

This recaptures the osaka replays using the altered JSON encoding for
replay files.

DavePearce commented on arithmetization/src/test/resources/replays/osaka/oogModexp.json:
```diff
@@ -0,0 +1 @@
+{"jsonrpc":"2.0","id":1,"result":{"capture":"{\"fork\":\"LONDON\",\"blocks\":[{\"header\":{\"parentHash\":\"0x132a1775404b2c42fc89e27043e0a5ad7857fb1dc3b1b7bb436d9a7bf5c811bf\",\"ommersHash\":\"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347\",\"coinbase\":\"0x0000000000000000000000000000000000000000\",\"stateRoot\":\"0x811836250d111695914e7cf47701421c4e6a7f2a1e3f667cac166d606e ...
```
Fixed

DavePearce commented on arithmetization/src/test/java/net/consensys/linea/ReplayTestTools.java:
```diff
@@ -89,8 +86,14 @@ public static void replay(
-        gson.fromJson(new BufferedReader(new InputStreamReader(stream)), ConflationSnapshot.class);
+    ;
+
+    final String conflationAsString;
```
Fixed

DavePearce reviewed (approved):
LGTM

DavePearce commented on arithmetization/src/test/java/net/consensys/linea/ReplayTestTools.java:
```diff
@@ -89,7 +85,11 @@ public static void replay(
 
     final Gson gson = new Gson();
-    final ConflationSnapshot conflation =
+    ;
```
fixed --- well spotted ;)

letypequividelespoubelles committed:
Merge branch '2538-add-one-osaka-replay-tests' of github.com:Consensys/linea-tracer into 2538-add-one-osaka-replay-tests

letypequividelespoubelles committed:
ras

Signed-off-by: F Bojarski <ceciestunepoubelle@protonmail.ch>

letypequividelespoubelles committed:
Merge branch 'arith-dev' into 2538-add-one-osaka-replay-tests

DavePearce committed:
remove ExempleTest
